MetaMask/metamask-extension
# feat(sentry): Adding extensionId and installType to Sentry

## **Description**
We want to differentiate between errors from our published extension, and errors from side-loaded extensions or forks that we don't support. Adding `extensionID` to our Sentry logs helps us do that. Following @Gudahtt's suggestion to use https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/management/getSelf to add `installType` as well.


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26482?quickstart=1)

## **Related issues**

Fixes: #26444 

## **Manual testing steps**

1. Run app, go to network tab in devtools
2. Filter for sentry envelope request and verify

## **Screenshots/Recordings**



### **Before**



### **After**


<img width="1728" alt="image" src="https://github.com/user-attachments/assets/9427b0f7-f1a0-4305-b16e-c48e3ec8e408">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
